pkg/auth/authz_handler.go

```diff
@@ -105,3 +119,44 @@ func OrgIDFromContext(ctx context.Context) string {
 	payload := PayloadFromContext(ctx)
 	return payload.Organization
 }
+
+func (a *AuthzHandler) isAuthorizedToAccessRoute(
+	request *http.Request,
+	payload *ocm.AuthPayload) bool {
+
+	route := middleware.MatchedRouteFrom(request)
+	if route.Authenticators == nil {
+		a.log.Debugf(
+			"%s: Authorized user: %s all scopes are allowed",
+			route.PathPattern, payload.Username)
+		return true
+	}
+
+	authScheme := route.Authenticator.Schemes[0]
+	for _, policy := range route.Authenticators {
+		policyScopes, exist := policy.Scopes[authScheme]
+		if !exist {
+			continue
+		}
+		if len(policyScopes) == 0 {
+			a.log.Debugf(
```

Comment:
not sure I understand. if I get it right I need to iterate over the policies twice - one time for the isPolicyScopesAvailable function, and other time here when comparing scopes. is this what you meant?

Reply:
Yes, that's why I wondered if we really need the loop, as we only use a single authenticator.
Maybe extracting only the logging to a separate func would make it a bit clearer (just to avoid duplication and make the func shorter)

Reply:
done